jkind/src/jkind/analysis/StaticAnalyzer.java

```diff
@@ -55,8 +55,17 @@ private static void checkErrors(Program program, SolverOption solver) {
 		valid = valid && propertiesBoolean(program);
 		valid = valid && ivcUnique(program);
 		valid = valid && ivcLocalOrOutput(program);
-		if (solver != SolverOption.Z3) {
-			valid = valid && LinearChecker.check(program, Level.ERROR);
+		
+		switch(solver) {
```

Comment:
I would change this around and whitelist the solvers that are okay to do non-linear with